File: back/models/address.go
package models

import "gorm.io/gorm"

type Address struct {
	gorm.Model
	Street    string   `json:"street"`
	Number    string   `json:"number"`
	City      string   `json:"city"`
	Zip       string   `json:"zip"`
	Latitude  *float64 `json:"latitude"`
	Longitude *float64 `json:"longitude"`
}

type AddressCreate struct {
	Street    string   `json:"street" validate:"required"`
	Number    string   `json:"number" validate:"required"`
	City      string   `json:"city" validate:"required"`
	Zip       string   `json:"zip" validate:"required"`
	Latitude  *float64 `json:"latitude"`
	Longitude *float64 `json:"longitude"`
}

func (e AddressCreate) ToAddress() *Address {
	return &Address{
		Street:    e.Street,
		Number:    e.Number,
		City:      e.City,
		Zip:       e.Zip,
		Latitude:  e.Latitude,
		Longitude: e.Longitude,
	}
}

File: front/lib/core/models/address.dart
import 'package:latlong2/latlong.dart';

class Address {
  final int id;
  final String street;
  final String number;
  final String city;
  final String zip;
  final LatLng? latlng;

  Address({
    required this.id,
    required this.street,
    required this.number,
    required this.city,
    required this.zip,
    this.latlng,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['ID'],
      street: json['street'],
      number: json['number'],
      city: json['city'],
      zip: json['zip'],
      latlng: json['latitude'] != null && json['longitude'] != null
          ? LatLng(json['latitude'], json['longitude'])
          : null,
    );
  }

  String get fullAddress => '$number $street, $zip $city';

  Map<String, dynamic> toJson() {
    return {
      'ID': id,
      'street': street,
      'number': number,
      'city': city,
      'zip': zip,
      'latitude': latlng?.latitude,
      'longitude': latlng?.longitude,
    };
  }
}

class AddressCreate {
  final String street;
  final String number;
  final String city;
  final String zip;

  AddressCreate({
    required this.street,
    required this.number,
    required this.city,
    required this.zip,
  });

  factory AddressCreate.fromJson(Map<String, dynamic> json) {
    return AddressCreate(
      street: json['street'],
      number: json['number'],
      city: json['city'],
      zip: json['zip'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'street': street,
      'number': number,
      'city': city,
      'zip': zip,
    };
  }
}
